### PR TITLE
pythonPackages.pynwb: init at 1.3.0

### PR DESCRIPTION
**Pull request by tbenst**
###### Motivation for this change
add package for working with Neurodata stored in the NWB format

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/pynwb/default.nix [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    h5py
+    hdmf
+    numpy
+    pandas
+    python-dateutil
+  ];
+
+  meta = with lib; {
+    description = "Package for working with Neurodata stored in the NWB format";
+    homepage = https://github.com/NeurodataWithoutBorders/pynwb;
```
Comment (jonringer) [inline pkgs/development/python-modules/pynwb/default.nix:30]: rfc 45
```suggestion
    homepage = "https://github.com/NeurodataWithoutBorders/pynwb";
```

File: pkgs/development/python-modules/hdmf/default.nix [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+, h5py
+, numpy
+, scipy
+, pandas
+, ruamel_yaml
+}:
+
+buildPythonPackage rec {
+  pname = "hdmf";
+  version = "1.6.1";
```
Comment (jonringer) [inline pkgs/development/python-modules/hdmf/default.nix:13]:
```suggestion
  version = "1.6.1";
  disabled = isPy27;
```

```
builder for '/nix/store/k531c83irjwjf533ind83a5fcw8m9ln1-python2.7-hdmf-1.6.1.drv' failed with exit code 1; last 10 log lines:
                           ^
  SyntaxError: invalid syntax


  ----------------------------------------------------------------------
  Ran 27 tests in 0.001s

  FAILED (errors=27)
  Test failed: <unittest.runner.TextTestResult run=27 errors=27 failures=0>
  error: Test failed: <unittest.runner.TextTestResult run=27 errors=27 failures=0>
cannot build derivation '/nix/store/4w0dg349mqg3wb61nigibzg7523i3zkz-python2.7-pynwb-1.3.0.drv': 1 dependencies couldn't be built
```

File: pkgs/development/python-modules/pynwb/default.nix [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+, h5py
+, hdmf
+, numpy
+, pandas
+, python-dateutil
+}:
+
+buildPythonPackage rec {
+  pname = "pynwb";
+  version = "1.3.0";
```
Comment (jonringer) [inline pkgs/development/python-modules/pynwb/default.nix:13]: this doesnt seem to build'
```
builder for '/nix/store/rxb3ydcy00gfq2gzbm3ivddvhj1fv22z-python3.8-pynwb-1.3.0.drv' failed with exit code 1; last 10 log lines:

  - ('hdmf-common', 'core', 'B2BL0UA4UN', '2X3ZCGXIMI')
  + ('hdmf-common', 'core')

  ----------------------------------------------------------------------
  Ran 284 tests in 39.757s

  FAILED (failures=1)
  Test failed: <unittest.runner.TextTestResult run=284 errors=0 failures=1>
  error: Test failed: <unittest.runner.TextTestResult run=284 errors=0 failures=1>
builder for '/nix/store/pskzl4mvapjzy754v0armf0rk6a1nh30-python3.7-pynwb-1.3.0.drv' failed with exit code 1; last 10 log lines:

  - ('hdmf-common', 'core', 'YDGT6MVJ98', '5Q5HEGTMFJ')
  + ('hdmf-common', 'core')

  ----------------------------------------------------------------------
  Ran 284 tests in 45.498s

  FAILED (failures=1)
  Test failed: <unittest.runner.TextTestResult run=284 errors=0 failures=1>
  error: Test failed: <unittest.runner.TextTestResult run=284 errors=0 failures=1>
cannot build derivation '/nix/store/qr9p8dxialvry4ipi2aqb37104palxnl-env.drv': 4 dependencies couldn't be built
[4 built (3 failed), 4 copied (1.4 MiB), 0.3 MiB DL]
error: build of '/nix/store/qr9p8dxialvry4ipi2aqb37104palxnl-env.drv' failed
https://github.com/NixOS/nixpkgs/pull/84857
4 package failed to build:
python27Packages.hdmf python27Packages.pynwb python37Packages.pynwb python38Packages.pynwb
```

**Comment by SuperSandro2000** [comment]
Closing due to inactivity from author.

**Pull request closed without merging**